packages/navie/test/services/completion-service-factory.spec.ts

```diff
@@ -12,15 +15,41 @@ describe('CompletionServiceFactory', () => {
 
   it('creates an OpenAICompletionService by default', () => {
     process.env.OPENAI_API_KEY = 'openai-key';
-    const trajectory: Trajectory = jest.genMockFromModule<Trajectory>('../../src/lib/trajectory');
+    const trajectory: Trajectory = jest.createMockFromModule<Trajectory>(
+      '../../src/lib/trajectory'
+    );
     const service = createCompletionService({ modelName: 'test', temperature: 1, trajectory });
     expect(service).toBeInstanceOf(OpenAICompletionService);
   });
 
   it('creates an AnthropicCompletionService when ANTHROPIC_API_KEY is set', () => {
     process.env.ANTHROPIC_API_KEY = 'anthropic-key';
-    const trajectory: Trajectory = jest.genMockFromModule<Trajectory>('../../src/lib/trajectory');
+    const trajectory: Trajectory = jest.createMockFromModule<Trajectory>(
+      '../../src/lib/trajectory'
+    );
     const service = createCompletionService({ modelName: 'test', temperature: 1, trajectory });
     expect(service).toBeInstanceOf(AnthropicCompletionService);
   });
+
+  it("abides by the selected model's provider", () => {
+    process.env.ANTHROPIC_API_KEY = 'anthropic-key';
+    const trajectory: Trajectory = jest.createMockFromModule<Trajectory>(
+      '../../src/lib/trajectory'
+    );
+    const examples: (NavieModel & { expected: unknown })[] = [
+      { id: 'gpt-4o', provider: 'openai', expected: OpenAICompletionService },
+      { id: 'claude-2', provider: 'anthropic', expected: AnthropicCompletionService },
+      { id: 'deepseek-r1:8b', provider: 'ollama', expected: OllamaCompletionService },
+      { id: 'gemini-1.5-turbo', provider: 'vertex-ai', expected: GoogleVertexAICompletionService },
+    ];
```

Comment:
I think you can avoid explicitly typing this by saying `const examples = [ /* ... */ ] as const;`. (Then you shouldn't need the NavieModel import.)